### ISSUE TITLE
Text after ) will be included in the last enum value

### ISSUE THREAD

**Issue by dsonck92** [abice/go-enum#102]
When adhering to godot linter rules, I must add a trailing `.` to the end of the `ENUM(...)`. But this will result in `).` to be added to the string name:

```go
// ENUM(1).
type Test int8
```
generates
```go
const (
	// Test1 is a Test of type 1)..
	Test1 Test = iota
)

const _TestName = "1)."
```

But even worse, on a multiline variant, it will add an empty item, colliding with the type:

```go
// ENUM(
//   X
//   Y
//   Z
// ).
type Test int8
```
generates
```go
const (
	// TestX is a Test of type X.
	TestX Test = iota
	// TestY is a Test of type Y.
	TestY
	// TestZ is a Test of type Z.
	TestZ
	// Test is a Test of type )..
	Test
)

const _TestName = "XYZ)."
```

I tried (to no avail) to disable godot for the ENUM line, since I find it a bit pointless. However, this does mean that it's also not possible to add some text behind the closing `)` which may be useful. Especially considering `ENUM(...values...)` kinda implies that parsing stops at `)`

**Comment by abice** [comment]
Man, good catch on that.

I'll start looking at it... In the mean time, you can always abide by Godot painting by writing a comment _before_ the ENUM declaration begins, and that should take care of that part of the problem.

Thanks for reporting this!

**Comment by dsonck92** [comment]
Actually, godot considers the ENUM() part of the comment. But the workaround is to put a dot behind the ENUM declaration on its own line.

Also the cause is quite simple, there's a `strings.TrimSuffix(xyz, ")")` which does not trim the trailing `)` as it's now trailing `).`. So perhaps some small state system detecting `ENUM(` as start token, and bailing out once `)` is found. Or a regex matching `ENUM([^)]?)` (might need some tweaking to handle multiline).

**Comment by dsonck92** [comment]
Yes, agreed on the first line needing to be a sentence. Curiously enough godot doesn't enforce that, it just has to have a dot all the way at the end. I guess, godot cannot detect the `ENUM()` part not being actual documentation, and it would be hard to let it to do so (unless some well defined generator options are used, I believe there is such a spec/convention but not sure, kubernetes controller code generators come to mind).

Anyways, it's fixed here, so, that's all there is to it. Thanks!

**Comment by dsonck92** [comment]
Seems like an edgecase was still going wrong but I've fixed that in #111
